File: library/src/androidTest/mock/com/mobilesolutionworks/android/app/test/basic/TestWorksFragment.java
package com.mobilesolutionworks.android.app.test.basic;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.mobilesolutionworks.android.app.WorksFragment;
import com.mobilesolutionworks.android.app.controller.WorksControllerManager;
import com.mobilesolutionworks.android.app.test.GetController;

/**
 * Created by yunarta on 9/3/17.
 */

public class TestWorksFragment extends WorksFragment implements GetController<TestWorksController> {

    private TestWorksController mController;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mController = getControllerManager().initController(0, null, new WorksControllerManager.ControllerCallbacks<TestWorksController>() {
            @Override
            public TestWorksController onCreateController(int id, Bundle args) {
                return new TestWorksController();
            }
        });
    }

    public TestWorksController getController() {
        return mController;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(com.mobilesolutionworks.android.app.test.R.layout.empty_fragment, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView id = (TextView) view.findViewById(com.mobilesolutionworks.android.app.test.R.id.textView);
        id.setText("empty");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mController = null;
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/basic/TestWorksFragment.kt
package com.mobilesolutionworks.android.controller.test.basic

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.mobilesolutionworks.android.app.WorksFragment
import com.mobilesolutionworks.android.app.controller.CreateCallback2
import com.mobilesolutionworks.android.controller.test.GetController
import com.mobilesolutionworks.android.controller.test.R

/**
 * Created by yunarta on 9/3/17.
 */

class TestWorksFragment : WorksFragment(), GetController<TestWorksController> {

    override var controller: TestWorksController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        controller = controllerManager.initController(0, null, CreateCallback2 { TestWorksController(it) })
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.empty_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val id = view.findViewById(R.id.textView) as TextView
        id.text = "empty"
    }

    override fun onDestroy() {
        super.onDestroy()
        controller = null
    }
}
